### PR TITLE
Organization: Users list: add organization selector

### LINKED ISSUE #303: A parent organization must list and manage children assets
=> Une structure parente doit pouvoir lister et modifier les véhicules de ses structures enfants
=> Ajouter un bouton "Véhicules" dans chaque ligne de la liste des structures enfants
=> Afficher tous les véhicules des enfants dans la liste des véhicules du parent, en ajouter la colonne "Structure"

### PR DESCRIPTION
**Pull request by nikophil**
fixes #303

### CODE REVIEW

File: translations/messages.fr.yaml [hunk 1]
```diff
@@ -94,6 +94,8 @@ organization:
   users: Bénévoles
   planning:
     no_resource_available: Aucune resource ne correspond à votre recherche
+  children_selector:
+    label: Organization
```
Comment (mRoca) [inline translations/messages.fr.yaml:98]: Structure

File: src/Repository/OrganizationRepository.php [hunk 2]
```diff
@@ -106,6 +106,14 @@ public function findByParentQueryBuilder(Organization $organization): QueryBuild
         return $qb;
     }
 
+    public function findChildrenQueryBuilder(Organization $organization): QueryBuilder
```
Comment (mRoca) [inline src/Repository/OrganizationRepository.php:109]: Duplicate of `findByParentQueryBuilder`

File: src/Form/Type/OrganizationSelectorType.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            ->add(
+                'organization',
+                EntityType::class,
+                [
+                    'class' => Organization::class,
+                    'label' => 'organization.children_selector.label',
+                    'query_builder' => $this->organizationRepository->findChildrenQueryBuilder($options['currentOrganization']),
+                    'choice_attr' => function (Organization $choice) use ($options) {
+                        return [
+                            'data-url' => $this->router->generate($options['route_to_redirect'], ['id' => $choice->getId()]),
```
Comment (mRoca) [inline src/Form/Type/OrganizationSelectorType.php:45]: very good idea, but we can only use it if the `id` route parameter refers to an organization

Comment (nikophil) [reply src/Form/Type/OrganizationSelectorType.php:45]: is that really a problem? for now we will only use that in `UserListController` and `ComissionnableAssetsListController` where this parameter exists

File: src/Controller/Organization/CommissionableAsset/CommissionableAssetDeleteController.php [hunk 4]
```diff
@@ -25,21 +27,21 @@ public function __construct(CommissionableAssetAvailabilityRepository $commissio
         $this->commissionableAvailabilityRepository = $commissionableAssetAvailabilityRepository;
     }
 
-    public function __invoke(EntityManagerInterface $entityManager, CommissionableAsset $commissionableAsset): RedirectResponse
+    public function __invoke(EntityManagerInterface $entityManager, CommissionableAsset $asset): RedirectResponse
     {
         $organization = $this->getUser();
         if (!$organization instanceof Organization || false === $organization->isParent()) {
```
Comment (mRoca) [inline src/Controller/Organization/CommissionableAsset/CommissionableAssetDeleteController.php:33]: this check can be replaced by a security check, maybe in another PR ?

Comment (nikophil) [reply src/Controller/Organization/CommissionableAsset/CommissionableAssetDeleteController.php:33]: Yep. this PR ils Big enough
I was also not sure about the rule: only parents can delete an asset?

File: src/Controller/Organization/CommissionableAsset/CommissionableEditAssetController.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+use App\Security\Voter\CommissionableAssetVoter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/{organization}/commissionable-assets/{asset}/edit", name="app_organization_commissionable_edit_asset", methods={"GET", "POST"}, requirements={"id": "\d+"})
```
Comment (fmeynard) [inline src/Controller/Organization/CommissionableAsset/CommissionableEditAssetController.php:18]: requirements should also be updated as "id" parameter is no longer used

**Pull request merged**